### PR TITLE
Exempts all medical staff from Doctor's Delight hunger drain by adding a new medical_metabolism mind trait.

### PR DESCRIPTION
**Pull request by CydiaLamiales**
## About The Pull Request
Adds a new mind trait - MEDICAL_METABOLISM and gives it to CMO, Medical Doctor, Chemist, Geneticist, Virologist, Brig Physician and Psychiatrist . 
Also makes the Doctor's Delight's hunger drain ineffective on all the roles listed above as opposed to just Medical Doctor.

## Why It's Good For The Game
Brings Doctor's Delight in line with Quadruple Sec which does actually effect everyone in the Security department.
The new mind trait allows for more foods and drinks that have unique effects on the staff of the Medical department to be easily added.

## Testing Photographs and Procedure

<details>
<summary>Screenshots&Videos</summary>

testing
Made a Virologist and a Bartender have 300 of nutrition and then drink 50u of Doctor's Delight each, After a couple of seconds the Bartender's nutrition started decreasing.

</details>

## Changelog
:cl:
add: Added a new mind trait for the medical staff
tweak: Tweaked the Doctor's Delight to not drain hunger from all medical staff
/:cl:

### CODE REVIEW

File: code/modules/reagents/chemistry/reagents/drink_reagents.dm [hunk 1]
```diff
@@ -718,7 +718,7 @@
 	M.adjustToxLoss(-0.5, 0)
 	M.adjustOxyLoss(-0.5, 0)
 	if(M.nutrition && (M.nutrition - 2 > 0))
-		if(M.mind?.assigned_role != JOB_NAME_MEDICALDOCTOR) //Drains the nutrition of the holder. Not medical doctors though, since it's the Doctor's Delight!
+		if(M.mind && !HAS_TRAIT(M.mind, TRAIT_MEDICAL_METABOLISM)) //Drains the nutrition of the holder. Not medical staff though, since it's the Doctor's Delight!
```
Comment (Penwin0) [inline code/modules/reagents/chemistry/reagents/drink_reagents.dm:721]: Just check if its in GLOB.medical_positions, you really don't need a mind trait for this

**Review by Penwin0 — CHANGES_REQUESTED** [review]

**Comment by itsmeow** [comment]
PR title is badly worded. It sounds like the drink is draining the hunger of the medical department, as a whole, actively.

Something like "Exempt all medical staff from Doctor's Delight hunger drain"

Same from the CL.

File: code/modules/reagents/chemistry/reagents/drink_reagents.dm (hunk 1 above)
Comment (CydiaLamiales) [reply code/modules/reagents/chemistry/reagents/drink_reagents.dm:721]: GLOB.medical_positions doesn't include the Psychiatrist sadly. Also I want it to function like security having LAW_ENFORCEMENT_METABOLISM and all the foods having special effects for sec. Plus, this way you get to keep the benefits even if you change your job.

Comment (Penwin0) [reply code/modules/reagents/chemistry/reagents/drink_reagents.dm:721]: Fair enough then aye, feel free to resolve this

**Comment by CydiaLamiales** [comment]
Right, sorry, completely forgot about testing evidence. Added a screenshot to the description.

**Review by itsmeow — APPROVED** [review]

**Review by PowerfulBacon — APPROVED** [review]

**Pull request merged**